src/ophys_etl/decrosstalk/ophys_plane.py

```diff
@@ -0,0 +1,848 @@
+
+        motion_border -- dict defining the border of the valid region
+        within each frame, e.g.
+            {
+                "y1": 19.0758,
+                "y0": 22.3232,
+                "x0": 9.16988,
+                "x1": 7.79272
+             }
+        """
+
+        self._path = movie_path
+        self._motion_border = copy.deepcopy(motion_border)
+
+        # this is where the data from the movie file will be stored
+        self._data = None
+
+    @property
+    def path(self):
+        return self._path
+
+    @property
+    def motion_border(self):
+        return copy.deepcopy(self._motion_border)
+
+    def load_movie_data(self):
+        """
+        Load the data from self._path; store te data in self._data
+        """
+        with h5py.File(self.path, mode='r') as in_file:
+            self._data = in_file['data'][()]
+
+    @property
+    def data(self):
+        if self._data is None:
+            self.load_movie_data()
+        return self._data
+
+    def get_trace(self, roi_list):
+        """
+        Extract the traces from a movie as defined by the ROIs in roi_list
+
+        Parameters
+        ----------
+        roi_list -- a list of OphysROI instantiations
+                    specifying the ROIs from which to
+                    extract traces
+
+        Returns
+        -------
+        output -- a dict such that
+
+            output['roi'][roi_id] = np.array of trace values for the ROI
+
+            output['neuropil'][roi_id] = np.array of trace values defined
+                                         in the neuropil around the ROI
+        """
+        motion_border = [self._motion_border['x0'], self._motion_border['x1'],
+                         self._motion_border['y0'], self._motion_border['y1']]
+
+        height = self.data.shape[1]
+        width = self.data.shape[2]
+
+        roi_mask_list = []
+        for roi in roi_list:
+            pixels = np.argwhere(roi.mask_matrix)
+            pixels[:, 0] += roi.y0
+            pixels[:, 1] += roi.x0
+            mask = roi_masks.create_roi_mask(width, height, motion_border,
+                                             pix_list=pixels[:, [1, 0]],
+                                             label=str(roi.roi_id),
+                                             mask_group=-1)
+
+            roi_mask_list.append(mask)
+
+        _traces = roi_masks.calculate_roi_and_neuropil_traces(self.data,
+                                                              roi_mask_list,
+                                                              motion_border)
+        roi_traces = _traces[0]
+        neuropil_traces = _traces[1]
+
+        output = {}
+        output['roi'] = {}
+        output['neuropil'] = {}
+        for i_roi, roi in enumerate(roi_list):
+            output['roi'][roi.roi_id] = roi_traces[i_roi]
+            output['neuropil'][roi.roi_id] = neuropil_traces[i_roi]
+        return output
+
+
+class OphysPlane(object):
```

Comment:
This class gives me pause. First, it's name is very general, making me think it is a general data structure for a plane. But, then it has methods specific to decrosstalk. I'd be more inclined to define strict data-only structures and stand-alone methods that accept and return those data structures.
We get a lot of classes like this from the science team and they can be tough to test and maintain because somewhere deep in a method there is a `self.some_important_parameter` that is set by `self.some_other_method()` which isn't very transparent.
I understand it is a big ask to refactor this with time constraints, but, something to keep in mind later.

Reply:
Really, the `OphysPlane` class just exists to keep and `OphysMovie` and a list of `OphysROIs` together. You are correct that all of the processing methods could be pulled out of the class, as long as `run_decrosstalk` accepts two `OphysPlane` instantiations.

How would you feel about renaming the class `DecrosstalkingPlane` and then refactoring the methods as described (`run_decrosstalk` accepts two `DecrosstalkingPlane`s; everything else remains the same, but outside of any class)?

Reply:
@njmei
I'm pinging you because this initial design was meant to satisfy your request for something that looked like
```
plane1 = OphysPlane(data1)
plane2 = OphysPlane(data2)

plane1.run_decrosstalk(plane2)
plane2.run_decrosstalk(plane1)
```

I assume something that looks like

```
plane1 = OphysPlane(data1)
plane2 = OphysPlane(data2)

run_decrosstalk(plane1, plane2)
run_decrosstalk(plane2, plane1)
```
won't offend your sensibilities?

Reply:
@danielsf yes, that works for me.

Reply:
I'd feel better about it, but this isn't a showstopper for me. The class is not too gnarly, it is just a style that eventually gets gnarly, if persisted.

Reply:
About to push the fix